src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNVectorFieldMapper.java

```diff
@@ -56,38 +57,48 @@
 
     protected Explicit<Boolean> ignoreMalformed;
 
-    public KNNVectorFieldMapper(String simpleName, MappedFieldType fieldType, MappedFieldType defaultFieldType,
+    public KNNVectorFieldMapper(String simpleName, FieldType fieldType, MappedFieldType defaultFieldType,
                                 Settings indexSettings, MultiFields multiFields, Explicit<Boolean> ignoreMalformed,
                                 CopyTo copyTo) {
-        super(simpleName, fieldType, defaultFieldType, indexSettings, multiFields, copyTo);
+        super(simpleName, fieldType, defaultFieldType,  multiFields, copyTo);
         this.ignoreMalformed = ignoreMalformed;
     }
 
     public static class Names {
         public static final String IGNORE_MALFORMED = "ignore_malformed";
     }
 
+    public KNNVectorFieldMapper clone() {
+        return (KNNVectorFieldMapper) super.clone();
+    }
+
     public static class Defaults {
         public static final Explicit<Boolean> IGNORE_MALFORMED = new Explicit<>(false, false);
-        public static final KNNVectorFieldType FIELD_TYPE = new KNNVectorFieldType();
```

Comment:
Why was KNNVectorField removed?

Reply:
There is a refactoring around MappedFieldType and FieldType separating indexing and search attributes. More details here https://github.com/elastic/elasticsearch/pull/58160. As part of this i need to replace the MappedFieldType to FieldType for providing the indexing attributes.